sources/Interop/Hid/shared/hidpi/Hid.Manual.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+// Ported from shared/hidpi.h in the Windows SDK for Windows 10.0.18362.0
+// Original source is Copyright © Microsoft. All rights reserved.
+
+using System.Runtime.CompilerServices;
+
+namespace TerraFX.Interop
+{
+    public static unsafe partial class Hid
+    {
+        public const int FACILITY_HID_ERROR_CODE = 0x11;
+
+        public const ushort HIDP_LINK_COLLECTION_ROOT = unchecked((ushort)-1);
+
+        public const ushort HIDP_LINK_COLLECTION_UNSPECIFIED = 0;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool HidP_IsSameUsageAndPage(USAGE_AND_PAGE* u1, USAGE_AND_PAGE* u2) => *(uint*)u1 == *(uint*)u2;
+
+        public static int HidP_SetButtons(HIDP_REPORT_TYPE ReportType, ushort UsagePage, ushort LinkCollection, ushort* UsageList, uint* UsageLength, HIDP_PREPARSED_DATA* PreparsedData, sbyte* Report, uint ReportLength) =>
+            HidP_SetUsages(ReportType, UsagePage, LinkCollection, UsageList, UsageLength, PreparsedData, Report, ReportLength);
+
+        public static int HidP_UnsetButtons(HIDP_REPORT_TYPE ReportType, ushort UsagePage, ushort LinkCollection, ushort* UsageList, uint* UsageLength, HIDP_PREPARSED_DATA* PreparsedData, sbyte* Report, uint ReportLength) =>
+            HidP_UnsetUsages(ReportType, UsagePage, LinkCollection, UsageList, UsageLength, PreparsedData, Report, ReportLength);
+
+        public static int HidP_GetButtons(HIDP_REPORT_TYPE ReportType, ushort UsagePage, ushort LinkCollection, ushort* UsageList, uint* UsageLength, HIDP_PREPARSED_DATA* PreparsedData, sbyte* Report, uint ReportLength) =>
+            HidP_GetUsages(ReportType, UsagePage, LinkCollection, UsageList, UsageLength, PreparsedData, Report, ReportLength);
+
+        public static int HidP_GetButtonsEx(HIDP_REPORT_TYPE ReportType, ushort LinkCollection, USAGE_AND_PAGE* ButtonList, uint* UsageLength, HIDP_PREPARSED_DATA* PreparsedData, sbyte* Report, uint ReportLength) =>
+            HidP_GetUsagesEx(ReportType, LinkCollection, ButtonList, UsageLength, PreparsedData, Report, ReportLength);
+
+        public static int HIDP_ERROR_CODES(int sev, int code) => ((sev) << 28) | (FACILITY_HID_ERROR_CODE << 16) | code;
+
+        public static readonly int HIDP_STATUS_SUCCESS = HIDP_ERROR_CODES(0x0, 0);
```

Comment:
For these, the intent is that they are a `const`, similarly to how I did the errors in `winerror.h` even though they are defined using the various `MAKE_HRESULT` macros

Reply:
Are you referring to these?
```cs
public static int MAKE_D2DHR(int sev, int code) => MAKE_HRESULT(sev, FACILITY_D2D, code);
```
I see the only difference is field vs. property. Does the `=>` lead to the compiler treating this as a constant somehow? I'm not sure what it is I should fix.

Reply:
No, the member would have to be redeclared as a `public const int HIDP_STATUS_SUCCESS = resultOfMacro` rather than being a `static readonly` that calls a helper method.

Initializing all these statics has a cost and prevents some downstream optimizations and the C/C++ compile treats these type of macros as actual constants as well.
It's probably not something that is easily handled in the Powershell script and I've done it manually up to this point (as I haven't finished getting macro support added to ClangSharp).

Reply:
There aren't too many. I can do it manually for now.